File: MoreHighlight/src/main/java/xyz/wingio/plugins/morehighlight/TextNode.java
package xyz.wingio.plugins.morehighlight;

import com.discord.simpleast.core.node.Node;
import com.discord.utilities.textprocessing.*;

import android.text.SpannableStringBuilder;

public class TextNode<MessageRenderContext> extends Node<MessageRenderContext> {
  String content;

  public TextNode(String content){
    super();
    this.content = content;
  }

  @Override
  public void render(SpannableStringBuilder builder, MessageRenderContext renderContext) {
    builder.append(content);
  }
}

File: MoreHighlight/src/main/java/xyz/wingio/plugins/morehighlight/node/TextNode.kt
package xyz.wingio.plugins.morehighlight.node

import android.text.SpannableStringBuilder
import com.discord.simpleast.core.node.Node

class TextNode<MessageRenderContext>(
    private val content: String
): Node<MessageRenderContext>() {

    override fun render(builder: SpannableStringBuilder, renderContext: MessageRenderContext) {
        builder.append(content)
    }

}